### PR TITLE
Content generator: allow functional key patterns and more

### PR DESCRIPTION
**Pull request by snazy**
Adds ability to generate content keys in the content generator tool derived from a new key-generator, with support for random UUIDs, random strings, random integers or integer sequence. Ability to trigger a "next value" either always, after N uses or based on a probability.

Also enhances the content generator tool to optionally continue on errors (conflicts), and add more than one put operation to every commit.

### CODE REVIEW

**Review by dimas-b — COMMENTED** [review]
LGTM overall. Minor comments added.

File: tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/Functions.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                      "Nested function pattern did not resolve to a function"));
+    }
+  }
+
+  static final class UuidFunc implements Func {
+    UuidFunc() {}
+
+    @Override
+    public void apply(Random random, StringBuilder target) {
+      target.append(randomUUID());
```
[the remark this answers is not in the document]
Comment (snazy) [reply tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/Functions.java:111]: done

File: tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/PatternParser.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            state = State.DOLLAR;
+          } else {
+            currentConstant.append(c);
+          }
+          break;
+        case DOLLAR:
+          if (c == '{') {
+            state = State.FUNC;
+          } else if (c == '$') {
+            currentConstant.append('$');
```
[the remark this answers is not in the document]
Comment (snazy) [reply tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/PatternParser.java:74]: done

File: tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/Functions.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      long v = (random.nextLong() & Long.MAX_VALUE) % bound;
+      target.append(v);
+    }
+  }
+
+  static final class StringFunc implements Func {
+    private final int len;
+
+    private static final char[] CHARS =
+        ("ABCDEFGHIJKLMNOPQRSTUVWXYZ" + "abcdefghijklmnopqrstuvwxyz" + "0123456789" + "b._-")
```
[the remark this answers is not in the document]
Comment (snazy) [reply tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/Functions.java:134]: b/c space-bar key and 'b' key are too close

File: tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/Functions.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+      return Long.parseLong(params.removeFirst());
+    }
+
+    static Func funcParam(Deque<String> params) {
+      String name = params.removeFirst();
+      return resolveFunction(name)
+          .generate(params)
+          .orElseThrow(
+              () ->
+                  new IllegalArgumentException(
```
[the remark this answers is not in the document]
Comment (snazy) [reply tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/Functions.java:101]: yes, if you write something like `${seq,20,fooooooooooo}`

[the remark this answers is not in the document]
Comment (dimas-b) [reply tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/keygen/Functions.java:101]: Hmmm... sorry, I do not see how that is possible. My reading is that we'll get a solid exception before we reach this place... Would you mind adding a test case for this?

**Review by dimas-b — APPROVED** [review]

**Pull request merged**